### PR TITLE
fix configuration issues

### LINKED ISSUE #601: [contracts] Compute all config rather than store in state
Currently throughout the app when we use a config value in models, we read the config then store that in the state of a model.

This hard codes the config value for all future interactions with this model.

We need to move this so all config values are computed on the fly, rather than hard coding when a model is created. This will reduce gas costs, and also allow the config to change within a session.

--------

This issue basically concerns the `CapacityConfig` and `SpeedConfig`. <br><br>
Regarding capacity configuration, you'll see that we hardcode the `Capacity` value of entities here
https://github.com/BibliothecaDAO/eternum/blob/cc2089f6632293d55cbbd5e11e470ca239f6f92e/contracts/src/systems/combat/contracts.cairo#L120

instead of doing that, we could store the entity type of in the `EntityMetadata` model
then whenever we need to get the capacity of the entity, we can get it by checking the current capacity config value of the entity type e.g
```
let entity_metadata: EntotyMetadata = get!(world, entity_id, EntityMetadata);
let capacity : CapacityConfig = CapacityConfigImpl::get(world, entity_metadata.entity_type);
```
then we use the capacity computed on the fly when determining the weight of entities
https://github.com/BibliothecaDAO/eternum/blob/cc2089f6632293d55cbbd5e11e470ca239f6f92e/contracts/src/models/weight.cairo#L10-L35
<br><br>

With regard to `SpeedConfig`, we will have to stop storing `ArrivalTime` and instead store the start time of a journey so that we can always calculate the arrival time on the fly by multiplying an entity's speed with the distance of journey
<br><br><br><br>

### LINKED ISSUE #1517: [contract] can't pause production
'u128_sub Overflow')

happening on both of my resource productions. 

image
image

### PR DESCRIPTION
**Pull request by credence0x**
- update capacity config so that capacity is calculated on the fly and can be updated
- removed storehouse capacity model and used general capacity model
- remove unused road and road config models
- update client and config to accomadate changes

resolves #1517 
resolves #601

### CODE REVIEW

**Review by aymericdelab — COMMENTED** [review]
looks like CapacityConfigCategory::Structure is not actually used anywhere in the contracts

**Comment by credence0x** [comment]
> looks like CapacityConfigCategory::Structure is not actually used anywhere in the contracts

it is used everywhere a structure is created e.g 
```
// bank
            set!(
                world,
                (
                    Structure {
                        entity_id: ADMIN_BANK_ENTITY_ID,
                        category: StructureCategory::Bank,
                        created_at: starknet::get_block_timestamp()
                    },
                    StructureCount { coord, count: 1 },
                    CapacityCategory { entity_id: ADMIN_BANK_ENTITY_ID, category: CapacityConfigCategory::Structure },
                    Bank { entity_id: ADMIN_BANK_ENTITY_ID, owner_fee_num, owner_fee_denom, exists: true },
                    Position { entity_id: ADMIN_BANK_ENTITY_ID, x: coord.x, y: coord.y },
                    Owner { entity_id: ADMIN_BANK_ENTITY_ID, address: admin }
                )
            );
            ```
            @aymericdelab

File: sdk/packages/eternum/src/constants/global.ts [hunk 1]
```diff
@@ -42,8 +42,18 @@ export const EternumGlobalConfig = {
...
   carryCapacityGram: {
-    donkey: 100_000,
-    army: 10_000,
+    [CapacityConfigCategory.None]: 0,
+    [CapacityConfigCategory.Structure]: BigInt(2) ** BigInt(128) - BigInt(1),
+    [CapacityConfigCategory.Donkey]: 100_000,
+    [CapacityConfigCategory.Army]: 10_000,
+    [CapacityConfigCategory.Storehouse]: 10_000_000_000,
+  },
+  carryCapacityKg: {
```
[the remark this answers is not in the document]
Comment (credence0x) [reply sdk/packages/eternum/src/constants/global.ts]: removed

**Comment by aymericdelab** [comment]
lgtm

**Pull request merged**